ignite/contrib/handlers/tqdm_logger.py

```diff
@@ -10,47 +10,83 @@ class ProgressBar:
 
+    Args:
+        persist (bool, optional): set to ``True`` to persist the progress bar after completion (default = ``False``)
+
     Examples:
 
-        Create a progress bar that shows you some metrics as they are computed,
-        by simply attaching the progress bar object to your engine.
+        Simple progress bar
+
+        .. code-block:: python
+
+            trainer = create_supervised_trainer(model, optimizer, loss)
+
+            pbar = ProgressBar()
+            pbar.attach(trainer)
+
+        Attach metrics that already have been computed at `ITERATION_COMPLETED` (such as `RunningAverage`)
 
         .. code-block:: python
 
+            trainer = create_supervised_trainer(model, optimizer, loss)
+
+            RunningAverage(output_transform=lambda x: x).attach(trainer, 'loss')
+
             pbar = ProgressBar()
             pbar.attach(trainer, ['loss'])
 
+        Directly attach the engine's output
+
+        .. code-block:: python
+
+            trainer = create_supervised_trainer(model, optimizer, loss)
+
+            pbar = ProgressBar()
+            pbar.attach(trainer, output_transform=lambda x: {'loss': x})
+
     Note:
         When adding attaching the progress bar to an engine, it is recommend that you replace
         every print operation in the engine's handlers triggered every iteration with
         ``pbar.log_message`` to guarantee the correct format of the stdout.
     """
 
-    def __init__(self):
+    def __init__(self, persist=False):
         self.pbar = None
+        self.persist = persist
 
     def _reset(self, engine):
         self.pbar = tqdm(
             total=len(engine.state.dataloader),
-            leave=False,
+            leave=self.persist,
             bar_format='{desc}[{n_fmt}/{total_fmt}] {percentage:3.0f}%|{bar}{postfix} [{elapsed}<{remaining}]')
 
     def _close(self, engine):
         self.pbar.close()
         self.pbar = None
 
-    def _update(self, engine, metric_names=None):
+    def _update(self, engine, metric_names=None, output_transform=None):
         if self.pbar is None:
             self._reset(engine)
 
         self.pbar.set_description('Epoch {}'.format(engine.state.epoch))
 
+        metrics = {}
         if metric_names is not None:
             if not all(metric in engine.state.metrics for metric in metric_names):
                 self._close(engine)
                 raise KeyError("metrics not found in engine.state.metrics")
 
-            metrics = {name: '{:.2e}'.format(engine.state.metrics[name]) for name in metric_names}
+            metrics.update({name: '{:.2e}'.format(engine.state.metrics[name]) for name in metric_names})
+
+        if output_transform is not None:
+            output_dict = output_transform(engine.state.output)
+
+            if not isinstance(output_dict, dict):
+                output_dict = {"output": output_dict}
+
+            metrics.update({name: '{:.2e}'.format(value) for name, value in output_dict.items()})
```

Comment:
Why is `{:.2e}` used here? Every _ignite_ examples uses the `{:.2f}` formatting, right?

Reply:
Exponential notation generalizes better regardless of the type of experiment. If you keep .2f you risk not having enough resolution to check the output

Reply:
Yes that true, that's why I use `{:.4f}` most of the time as I ike it more than the exponential notation. Maybe there should be a way to specify the format?

Reply:
@miguelvr I think we can change to `{:.4f}`. I would prefer.

Reply:
@vfdev-5 I currently have a project where even that resolution wouldn't cover one of my variables. I still vouch for the exponential notation as it is what guarantees resolution over all the cases.

Reply:
@miguelvr ok, let's keep it `{:.2e}` to guarantee the resolution.